src/pyobsplot/obsplot.py

```diff
@@ -319,3 +325,142 @@ def save_to_file(path: str, res: Union[SVG, HTML]) -> None:
             )
         with open(path, "w", encoding="utf-8") as f:
             f.write(str(res.data))
+
+
+class ObsplotTypstCreator(ObsplotJsdomCreator):
+    """
+    Jsdom renderer Creator class.
+    """
+
+    def __init__(
+        self,
+        theme: str = DEFAULT_THEME,
+        default: Optional[dict] = None,
+        debug: bool = False, 
+        font_size: int = 12,
+        font: str = "sans-serif",
+        dpi: int = 300,
+        margin: int = 4
+    ) -> None:
+        super().__init__(theme, default, debug)
+        self._proc = None
+        self.font_size = font_size
+        self.font = font
+        self.dpi = dpi
+        self.margin = margin
+        self.start_server()
+
+    def __call__(self, *args, **kwargs) -> None:
+        """
+        Method called when an instance is called.
+        """
+        if self._proc is not None and self._proc.poll() is not None:
+            msg = "Server has ended, please recreate your plot generator object."
+            raise RuntimeError(msg)
+        path = None
+        if "path" in kwargs:
+            path = kwargs["path"]
+            del kwargs["path"]
+        spec = self.get_spec(*args, **kwargs)
+        if "figure" not in spec:
+            spec["figure"] = True
+        res = ObsplotJsdom(
+            spec,
+            port=self._port,
+            theme=self._theme,
+            default=self._default,
+            debug=self._debug,
+        ).plot()
+        if path is None:
+            with tempfile.NamedTemporaryFile(suffix=".png") as f:
+                self.render_typst(str(res.data), f.name)
+                width = spec["width"] if "width" in spec else 640
+                if "margin" in spec:
+                    width += 2*spec["margin"]
+                else: 
+                    if "marginLeft" in spec:
+                        width += spec["marginLeft"]
+                    if "marginRight" in spec:
+                        width += spec["marginRight"]
+                display(
+                    Image(filename=f.name, width=width, height=spec["height"] if "height" in spec else None)
```

Comment:
Without this change you get a permission error on Windows. It should also be more efficient. See [here](https://stackoverflow.com/a/23212515). 


```suggestion
                    Image(data=f.read(), width=width, height=spec["height"] if "height" in spec else None)
```